vvm/utils/versioning.py

```diff
@@ -2,62 +2,60 @@
 import re
 from typing import Any, Optional
 
-from packaging.specifiers import Specifier
+from packaging.specifiers import SpecifierSet
 from packaging.version import Version
 
 from vvm.exceptions import UnexpectedVersionError
 from vvm.install import get_installable_vyper_versions, get_installed_vyper_versions
 
-_VERSION_RE = re.compile(r"\s*#\s*(?:pragma\s+|@)version\s+([=><^~]*)(\d+\.\d+\.\d+\S*)")
+_VERSION_RE = re.compile(r"^(?:#\s*(?:@version|pragma\s+version)\s+(.*))", re.MULTILINE)
 
 
-def _detect_version_specifier(source_code: str) -> Optional[Specifier]:
+def detect_version_specifier_set(source_code: str) -> Optional[SpecifierSet]:
     """
-    Detect the version given by the pragma version in the source code.
+    Detect the specifier set given by the pragma version in the source code.
 
     Arguments
     ---------
     source_code : str
-        Source code to detect the version from.
+        Source code to detect the specifier set from.
 
     Returns
     -------
-    str
-        vyper version specifier, or None if none could be detected.
+    Optional[SpecifierSet]
+        vyper version specifier set, or None if none could be detected.
     """
-    match = _VERSION_RE.search(source_code)
-    if match is None:
+    match = _VERSION_RE.findall(source_code)
+    if not match:
```

Comment:
i'm not a fan of using truthy in predicates -- how about `if len(matches) == 0:`

Reply:
reverted to search instead of findall, as there was no reason to use the latter here.